File: Human_written_code/Code_with_comments/CPP/Q29.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    cin >> t;  // Read the number of test cases

    while (t--) {
        cin >> n;
        int arr[n];
        
        for (int i = 0; i < n; i++) {
            cin >> arr[i];  // Read array elements
        }

        int ct = 0;
        sort(arr, arr + n);  // Sort the array to check for duplicates

        // Check if there are any consecutive duplicate elements
        for (int i = 0; i < n - 1; i++) {
            if (arr[i] == arr[i + 1]) {  // If duplicate found, output "NO" and break
                cout << "NO" << endl;
                break;
            } else {
                ct++;
            }
        }

        // If all elements are unique, output "YES"
        if (ct == n - 1)
            cout << "YES" << endl;
    }

    return 0;
}

File: Human_written_code/Code_with_comments/Python/Q29.py
def main():
    t = int(input().strip())  # Read the number of test cases

    for _ in range(t):
        n = int(input().strip())  # Read the number of elements in the array
        arr = set()  # Use a set to store unique elements

        duplicate_found = False
        for _ in range(n):
            inp = int(input().strip())  # Read each element
            if not duplicate_found:
                if inp not in arr:
                    arr.add(inp)  # Add to set if not seen before
                else:
                    duplicate_found = True  # Mark if a duplicate is found

        # Output "NO" if duplicate was found, otherwise "YES"
        print("NO" if duplicate_found else "YES")

if __name__ == "__main__":
    main()
